code/game/gamemodes/shadowling/shadow_abilities.dm

```diff
@@ -0,0 +1,542 @@
+	name = "Glare"
+	desc = "Disrupts the target's motor and speech abilities."
+	panel = "Shadowling Abilities"
+	charge_max = 300
+	human_req = 1
+	clothes_req = 0
+	action_icon_state = "glare"
+
+
+/obj/effect/proc_holder/spell/targeted/sling/glare/InterceptClickOn(mob/living/caller, params, atom/t)
+	. = ..()
+	if(target.stat)
+		to_chat(user, "<span class='warning'>[target] must be conscious!</span>")
+		revert_cast()
+		return
+	if(is_shadow_or_thrall(target))
+		to_chat(user, "<span class='warning'>You cannot glare at allies!</span>")
+		revert_cast()
+		return
+	var/mob/living/carbon/human/M = target
+	user.visible_message("<span class='warning'><b>[user]'s eyes flash a purpleish-red!</b></span>")
+	var/distance = get_dist(target, user)
+	if (distance <= 1) //Melee
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
+		target.Stun(100)
+		M.silent += 10
+	else //Distant glare
+		var/loss = 100 - ((distance - 1) * 18)
+		target.adjustStaminaLoss(loss)
+		target.stuttering = loss
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and exhaustion floods your body...</span>")
+		target.visible_message("<span class='danger'>[target] looks very tired...</span>")
+	charge_counter = 0
+	start_recharge()
+	remove_ranged_ability()
+	user = null
+	target = null
+
+
+////////////
+////VEIL////
+////////////
+
+/obj/effect/proc_holder/spell/veil //Puts out most nearby lights except for flares and yellow slime cores
+	name = "Veil"
+	desc = "Extinguishes most nearby light sources."
+	panel = "Shadowling Abilities"
+	charge_max = 150 //Short cooldown because people can just turn the lights back on
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon_state = "veil"
+	action_icon = 'icons/mob/actions/actions_shadowling.dmi'
+	var/blacklisted_lights = list(/obj/item/device/flashlight/flare, /obj/item/device/flashlight/slime)
+	var/admin_override = FALSE //Requested by Shadowlight213. Allows anyone to cast the spell, not just shadowlings.
+
+/obj/effect/proc_holder/spell/veil/cast(list/targets,mob/user = usr)
+	if(!shadowling_check(user) && !admin_override)
+		revert_cast()
+		return
+	to_chat(user, "<span class='shadowling'>You silently disable all nearby lights.</span>")
+	var/list/in_view = view(6, user)
+	for(var/turf/T in in_view)
+		for(var/obj/F in T.contents)
+			F.disable_lights()
+		for(var/mob/living/H in T.contents)
+			for(var/obj/item/F in H)
+				F.disable_lights()
+		for(var/obj/machinery/camera/cam in T.contents)
+			cam.set_light(0)
+			if(prob(10))
+				cam.emp_act(2)
+		for(var/mob/living/silicon/robot/borg in T.contents)
+			if(!borg.lamp_cooldown)
+				borg.update_headlamp(TRUE, INFINITY)
+				to_chat(borg, "<span class='danger'>Your headlamp is fried! You'll need a human to help replace it.</span>")
```

Comment:
I think this is fair for nightmares because nightmares have to land a hit. For shadowlings, this effectively makes borgs completely useless against shadowlings due to how low of a cooldown Veil has.